src/components/user-form/AddUserForm.js

```diff
@@ -1,103 +1,210 @@
-import React, { Component, createRef } from 'react';
+import React, {Component, createRef} from 'react';
 
 class AddUserForm extends Component {
 
-  firstNameRef = createRef();
-  lastNameRef = createRef();
-  emailNameRef = createRef();
-  // todo 1: добавить ref для address
-
-
-  state = {
-    warning: ''
-  };
-
-  onSubmit = (e) => {
-    // todo 1: достать из props функцию добавления пользователя onUserAdd
-    e.preventDefault();
-
-    console.log(this.firstNameRef.current.value);
-    console.log(this.lastNameRef.current.value);
-
-    const firstName = this.firstNameRef.current.value;
-    const lastName = this.lastNameRef.current.value;
-    const email = this.emailNameRef.current.value;
-
-    const pattern = /\d+/;
-
-    if (pattern.test(firstName) || pattern.test(lastName))  {
-      this.setState({
-        warning: 'please don\'t put numbers '
-      });
-      return;
+    firstNameRef = createRef();
+    lastNameRef = createRef();
+    emailNameRef = createRef();
+    // donetodo 1: добавить ref для address
+    addressNameRef = createRef()
+    //(gender, dob, phone,
+    // website, _links : {“avatar"})
+    genderNameRef = createRef()
+    dobNameRef = createRef()
+    phoneNameRef = createRef()
+    websiteNameRef = createRef()
+    linksAvatarNameRef = createRef()
+
+    state = {
+        warning: ''
+    };
+
+    onSubmit = (e) => {
+        // donetodo 1: достать из props функцию добавления пользователя onUserAdd
+        const {onUserAdd} = this.props
+        e.preventDefault();
+
+        console.log(this.firstNameRef.current.value);
+        console.log(this.lastNameRef.current.value);
+
+        const first_name = this.firstNameRef.current.value;
+        const last_name = this.lastNameRef.current.value;
+        const email = this.emailNameRef.current.value;
+        const address = this.addressNameRef.current.value;
+        const gender = this.genderNameRef.current.value;
+        const dob = this.dobNameRef.current.value;
+        const phone = this.phoneNameRef.current.value;
+        const website = this.websiteNameRef.current.value;
+        const linksAvatar = {avatar: this.linksAvatarNameRef.current.value}
+
+        const pattern = /\d+/;
+
+        if (pattern.test(first_name) || pattern.test(last_name)) {
+            this.setState({
+                warning: 'please don\'t put numbers '
+            });
+            return;
+        }
+
+        // console.log({firstName, lastName, email, address, gender, dob, phone, website, linksAvatar});
+        // donetodo 1: использовать функцию onUserAdd для создания нового пользователя здесь
+        //  у этого объекта должны быть проперти first_name, last_name, address, email
+
+        // onUserAdd && onUserAdd({first_name, last_name, email, address});
+        onUserAdd && onUserAdd({first_name, last_name, email, address, gender, dob, phone, website, linksAvatar});
```

Comment:
I would suggest to change data structure for storing avatar link
As I remember, it's more complex than just putting it on the highest level like linksAvatar

It should be stored in special object `_links: {avatar: { herf: ..... }}`

Please check constants/index.js to make sure that object structure is correct